src/Network/DogStatsd.hs

```diff
@@ -0,0 +1,53 @@
+module Network.DogStatsd (
+ dogStatsdClient
+, increment
+, decrement
+, count
+, gauge
+, timing
+, histogram
+) where
+
+import Network.Statsd (Stat, Type(..), fmtDatagram)
+import Network.Statsd.UdpClient(UdpClient(..), fromURI, send)
+
+import Control.Monad
+import Data.Maybe
+import Data.List
+import Data.Time.Units
+import Text.Printf
+import Network.URI
+
+type Tags = [(String, String)]
+
+dogStatsdClient :: String -> IO UdpClient
+dogStatsdClient url = (fromURI . fromJust . parseURI) url
```

Comment:
How does an error parsing the URI propagate here, if `parseURI = Nothing` `fromJust` will (forgive me I’ve forgotten the Haskell distinction between errors and exceptions) I think throw an exception?

Can we map that cast into `fail` in the `IO` monad? Should we? I think @joshvera mentioned that `fail` has moved into MonadFail in the latest GHC stdlib.

So long as this failure case is recoverable in an idiomatic Haskell fashion I’m :+1: